Framework/Built_In_Automation/Sequential_Actions/common_functions.py

```diff
@@ -7107,3 +7114,550 @@ def download_chrome_extension(data_set):
+    """Create a summary report from the raw axe results."""
+    summary = {
+        "test_info": {
+            "url": result.get("url", "Unknown"),
+            "timestamp": result.get("timestamp", "Unknown"),
+            "test_date": datetime.now().isoformat()
+        },
+        "summary": {
+            "violations_count": len(result.get("violations", [])),
+            "passes_count": len(result.get("passes", [])),
+            "inapplicable_count": len(result.get("inapplicable", [])),
+            "incomplete_count": len(result.get("incomplete", []))
+        },
+        "violations_summary": [],
+        "passes_summary": []
+    }
+
+    # Process violations
+    for violation in result.get("violations", []):
+        violation_summary = {
+            "id": violation.get("id", "Unknown"),
+            "description": violation.get("description", "No description"),
+            "impact": violation.get("impact", "Unknown"),
+            "help": violation.get("help", "No help available"),
+            "helpUrl": violation.get("helpUrl", ""),
+            "tags": violation.get("tags", []),
+            "nodes_count": len(violation.get("nodes", [])),
+            "nodes": []
+        }
+
+        # Add first few affected elements
+        for node in violation.get("nodes", [])[:3]:
+            node_info = {
+                "html": node.get("html", ""),
+                "target": node.get("target", []),
+                "failureSummary": node.get("failureSummary", "")
+            }
+            violation_summary["nodes"].append(node_info)
+
+        summary["violations_summary"].append(violation_summary)
+
+    # Process passes
+    for passed in result.get("passes", []):
+        pass_summary = {
+            "id": passed.get("id", "Unknown"),
+            "description": passed.get("description", "No description"),
+            "impact": passed.get("impact", "Unknown"),
+            "tags": passed.get("tags", []),
+            "nodes_count": len(passed.get("nodes", []))
+        }
+        summary["passes_summary"].append(pass_summary)
+
+    return summary
+
+#this function is used in the accessibilit_test action
+
+def safe_join_target(target):
+    """Safely join target field which can be a list or other types."""
+    try:
+        if not target:
+            return "No target"
+        if isinstance(target, list):
+            # Handle nested lists and complex structures
+            result = []
+            for item in target:
+                if isinstance(item, list):
+                    result.append(', '.join(str(x) for x in item))
+                else:
+                    result.append(str(item))
+            return ', '.join(result)
+        return str(target)
+    except Exception as e:
+        return f"Target error: {str(e)}"
+
+#this function is used in the accessibilit_test action
+
+def create_html_report(result, summary):
+    """Create a detailed HTML report from the raw axe results."""
+    html_template = f"""
```

Comment:
Don't generate html like this. Use jinja2 templating - this is already included as part of our Node.

Reply:
Also, the html template should be moved into it's own file.